code/__HELPERS/unsorted.dm

```diff
@@ -1671,34 +1671,50 @@ var/mob/dview/dview_mob = new
 //orbit() can run without it (swap orbiting for A)
 //but then you can never stop it and that's just silly.
 
-/atom/movable/proc/orbit(atom/A, radius = 10, clockwise = 1, angle_increment = 15)
+/atom/movable/proc/orbit(atom/A, radius = 10, clockwise = FALSE, rotation_speed = 20, rotation_segments = 36, pre_rotation = TRUE, lockinorbit = FALSE)
 	if(!istype(A))
 		return
-	orbiting = A
-	var/angle = 0
-	var/matrix/initial_transform = matrix(transform)
-	spawn
-		while(orbiting)
-			loc = orbiting.loc
 
-			angle += angle_increment
+	if(orbiting)
+		stop_orbit()
 
-			var/matrix/shift = matrix(initial_transform)
-			shift.Translate(radius,0)
-			if(clockwise)
-				shift.Turn(angle)
-			else
-				shift.Turn(-angle)
-			animate(src,transform = shift,2)
+	orbiting = A

```

Comment:
Did I not explicitly state that we shouldn't fucking port -tg-'s updated orbit proc?

This is way more complicated than it needs to be, and makes the damn thing much harder to implement anywhere.